blobby/blob.py

```diff
@@ -1,248 +0,0 @@
-from typing import BinaryIO, Callable, List, IO, Type, TypeVar, Iterator
-
-import hashlib
-import io
-import logging
-import os
-import tempfile
-import threading
-from contextlib import contextmanager
-from inspect import getmembers, isfunction, getdoc
-
-from mypy_boto3_s3.service_resource import Bucket as S3Bucket
-from botocore.response import StreamingBody
-
-logger = logging.getLogger(__name__)
-
-T = TypeVar('T')
-
-
-class BlobError(Exception):
-    pass
-
-
-def inherit_docstrings(cls: Type[T]) -> Type[T]:
-    for m in getmembers(cls):
-        if isfunction(m) and not m.__doc__:
-            m.__doc__ = getdoc(m)  # pragma: no cover
-
-    return cls
-
-
-@inherit_docstrings
-class BlobReader(io.RawIOBase):
-    """A read-only, file-like object wrapping an S3 blob."""
-    WHENCE_LIST = ', '.join(
-        str(t) for t in (io.SEEK_SET, io.SEEK_CUR, io.SEEK_END)
-    )
-
-    def __init__(self, s3_bucket: S3Bucket, key: str) -> None:
-        self.object = s3_bucket.Object(key=key)
-        self.position = 0
-
-    @contextmanager
-    def as_fifo(self) -> Iterator[BinaryIO]:
-        """Context manager which creates a FIFO and writes the blob to it.
-
-        Returns: an open file object of the read end of the FIFO, which
-            will be closed by the context manager.
-        """
-        with tempfile.TemporaryDirectory() as dir:
-            path = os.path.join(dir, 'fifo')
-            os.mkfifo(path, mode=0o600)
-            writer = threading.Thread(
-                target=self._fifo_writer, args=(path,)
-            )
-            writer.start()
-            with open(path, 'rb') as fd_r:
-                yield fd_r
-                writer.join()
-
-    def _fifo_writer(self, path: str) -> None:
```

Comment:
Semantic nitpick: the `er` suffix here implies that this is or is returning an object of some kind. Something like `_write_to_fifo` might be clearer. What do you think?

Reply:
Yeah, fair - I think since the only purpose is for it to be passed as a callable to `threading.Thread` when referenced it *is* a "do-er",  but since it's just a method it's not conventional to name it that way.